django/contrib/auth/backends.py

```diff
@@ -8,8 +8,8 @@ class ModelBackend(object):
     """
     Authenticates against settings.AUTH_USER_MODEL.
     """
-
-    def authenticate(self, username=None, password=None, **kwargs):
+    def authenticate(self, request, username=None, password=None, **kwargs):
+        self.request = request
```

Comment:
I'm not sure if this method should set an attribute as opposed to `authenticate()` passing `request` to the methods it calls as needed. @carljm, any thought about this?